nixos/tests/starship.nix

```diff
@@ -0,0 +1,42 @@
+import ./make-test-python.nix ({ pkgs, ... }: {
+  name = "starship";
+  meta.maintainers = pkgs.starship.meta.maintainers;
+
+  machine = {
+    programs = {
+      fish.enable = true;
+      zsh.enable = true;
+
+      starship = {
+        enable = true;
+        settings.format = "<starship>";
+      };
+    };
+
+    environment.systemPackages = map
+      (shell: pkgs.writeScriptBin "expect-${shell}" ''
+        #!${pkgs.expect}/bin/expect -f
+
+        spawn env TERM=xterm ${shell} -i
+
+        expect "<starship>" {
```

Comment:
Please use `installCheckPhase` to test this. See https://github.com/NixOS/nixpkgs/issues/143681

Besides that, LGTM

Reply:
personally wouldn't say this is an unnecessary VM test since it checks that the module has done the correct setup for shells to use the prompt. can that even be done from installCheck?

Reply:
The `expect` script could run in `installCheckPhase`; however that would only test that the package is working, not that the NixOS module has installed it correctly.

Reply:
I see, It makes sense keeping it as vm test then.

ping me after you resolve the conflicts to merge it :)